src/types-internal.h

```diff
@@ -263,6 +263,9 @@ struct netplan_parser {
     GHashTable* null_fields;
     GHashTable* null_overrides;
     GHashTable* global_renderer;
+
+    /* Flags used to change the parser's behavior */
+    unsigned int flags;
```

Comment:
suggestion (non-blocking): I'd love to see an additional boolean field that gets set whenever any parsing actually failed. It should get carrier over from NetplanParser into NetplanState, so we can check the final state, e.g. `State.parsing_errors` (not sure about the naming, yet).

Reply:
I like the idea. I just used a different approach, I added a counter to keep track of how many errors were ignored. I added it in two separate commits to make it easier to review.